File: repo/ruby/v2_translate.rb
# v2_translate.rb
# Instalar: gem install savon ruby-translate
# Ejecutar: ruby v2_translate.rb 10

require 'savon'
require 'ruby-translate'

WSDL = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL'

client   = Savon.client(wsdl: WSDL)
response = client.call(:number_to_words, message: { ubi_num: ARGV[0].to_i })
word_en  = response.body[:number_to_words_response][:number_to_words_result]

word_es  = RubyTranslate.translate(word_en, to: 'es', from: 'en')
puts word_es

File: repo/nodejs/v2_translate.js
// v2_translate.js
// Instalar: npm install
// Ejecutar: node v2_translate.js 10

const soap      = require('soap');
const translate = require('@vitalets/google-translate-api');

const WSDL = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL';
const num  = parseInt(process.argv[2]) || 10;

soap.createClient(WSDL, async (err, client) => {
    if (err) throw err;
    const [result] = await client.NumberToWordsAsync({ ubiNum: num });
    const en = result.NumberToWordsResult;
    const { text } = await translate(en, { from: 'en', to: 'es' });
    console.log(text);
});
